src/engine/thread.js

```diff
@@ -4,6 +4,12 @@
  */
 const _stackFrameFreeList = [];
 
+/**
+ * Default params object to for stack frames not inside a procedure.
```

Comment:
Nit: typo in this comment

Reply:
Maybe also explicitly call out the `Object.create(null)` here and add a note about why we're using it and that look-ups of params will not be able to use `hasOwnProperty`